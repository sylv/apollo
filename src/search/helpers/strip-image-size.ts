const VERSION_REGEX = /(?<=@\._V1_).*_(?=\.)/;
export const stripImageSize = (input: string) => {
  return input.replace(VERSION_REGEX, "");
};
