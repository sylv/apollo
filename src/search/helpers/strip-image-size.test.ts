import { expect, it } from "vitest";
import { stripImageSize } from "./strip-image-size";

it("should strip size info from urls", () => {
  expect(stripImageSize("https://m.media-amazon.com/images/M/MV5BNzQwNTA5NzA1MF5BMl5BanBnXkFtZTgwMjEwNzI0MjE@._V1_.jpg")).toMatchSnapshot();
  expect(
    stripImageSize("https://m.media-amazon.com/images/M/MV5BNzQwNTA5NzA1MF5BMl5BanBnXkFtZTgwMjEwNzI0MjE@._V1_UX224_CR0,0,224,126_AL_.jpg")
  ).toMatchSnapshot();
});
