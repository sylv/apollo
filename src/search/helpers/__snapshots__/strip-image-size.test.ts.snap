// Vitest Snapshot v1

exports[`should strip size info from urls 1`] = `"https://m.media-amazon.com/images/M/MV5BNzQwNTA5NzA1MF5BMl5BanBnXkFtZTgwMjEwNzI0MjE@._V1_.jpg"`;

exports[`should strip size info from urls 2`] = `"https://m.media-amazon.com/images/M/MV5BNzQwNTA5NzA1MF5BMl5BanBnXkFtZTgwMjEwNzI0MjE@._V1_.jpg"`;
